### PR TITLE
fix(semantic): remove inherting `ScopeFlags::Modifiers` from parent scope

### LINKED ISSUE #7900: Semantic: `ScopeFlags` for a class in a class constructor has `ScopeFlags::Constructor`
Input:

```js
class Outer {
  constructor() {
    class Inner {}
  }
}
```

Semantic sets `ScopeFlags` for `class Inner {}` as `StrictMode | Constructor`. I believe this is a mistake. It should just be `StrictMode`.

Weirdly I cannot replicate this in playground ([playground](https://playground.oxc.rs/#eNpFUEFOxDAM/ErlE0gVAiQufcBKnODAkUs2666CEruynYWq6t9xWnaRoiQzGc84XiDCADEH1e6tGkq3fFLXRSY1qdFY7u53yslN9UrUVGvjfPMFPTAMC0ilduhMFn5gGENW7CEnshvQyBPC4NYNzOXI+YpMAunIUv7Eaw9TEEXZPLlKxEPKSKG4ARiqPXypR+9PH3PzhcKnmtHZSdBLL/geBEn/LVs3m6Xfb4l7hgU5o7cKqM+PTy/gisgnPGP7lYOSKI3pWuwTMuF8yPzdii8oR1ZvYQ9a11+LTHim)). But on latest main, when I print `ScopeFlags` for all scopes I get this:

```
ScopeId(0): Program: ScopeFlags(StrictMode | Top)
ScopeId(1): Class(Outer): ScopeFlags(StrictMode)
ScopeId(2): Function(<anonymous>): ScopeFlags(StrictMode | Function | Constructor)
ScopeId(3): Class(Inner): ScopeFlags(StrictMode | Constructor)
```

Debug code:

```rs
let scopes = semantic.scopes();
let nodes = semantic.nodes();
#[allow(clippy::cast_possible_truncation)]
for i in 0..scopes.len() {
    let scope_id = ScopeId::new(i as u32);
    let flags = scopes.get_flags(scope_id);
    let node_id = scopes.get_node_id(scope_id);
    let kind = nodes.get_node(node_id).kind();
    let name = kind.debug_name();
    println!("{scope_id:?}: {name}: {flags:?}");
}
```

As this isn't visible on playground, is this the result of a very recent change? Or is there something else funny going on?

### PR DESCRIPTION
**Pull request by Dunqing**
close: #7900

After #4283 changed, we don't need to inherit `ScopeFlags` from the `constructor`, `set`, `get` anymore, I think this is a logic of forgetting to remove

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#7932** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7932?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7932?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=7932).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by Dunqing** [comment]
https://github.com/oxc-project/oxc/actions/runs/12347529516/job/34454785628?pr=7932
I just realized why we still need to inherit when I saw the failed linter test, But I think we should fix this test in that linter rule rather than by inheriting `ScopeFlags`.

File: crates/oxc_semantic/examples/semantic.rs [hunk 1]
```diff
@@ -39,6 +39,8 @@ fn main() -> std::io::Result<()> {
         .with_check_syntax_error(true)
         .build(&program);
 
+    println!("{:?}", semantic.semantic.scopes());
```
Comment (Boshen) [inline crates/oxc_semantic/examples/semantic.rs:42]:
```suggestion
```

File: crates/oxc_semantic/examples/semantic.rs [hunk 2]
```diff
@@ -39,6 +39,8 @@ fn main() -> std::io::Result<()> {
...
+
```
Comment (Boshen) [inline crates/oxc_semantic/examples/semantic.rs:43]:
```suggestion
```

**Review by overlookmotel — APPROVED** [review]

**Comment by overlookmotel** [comment]
### Merge activity

* **Dec 16, 6:25 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**